### DISCUSSION TITLE
Quick Q:Best way to migrate a mutable os workflow into podman.and how?

### DISCUSSION THREAD

**Discussion by devdailytester** [containers/podman#16583] [General]
TLDR:I have a dedicated setup for dev,(which is all in a folder with around GB size)

I originally use a mutable os like ubuntu and fedora workstation, in which you can play around with the home,path and etc,
I then moved to silverblue which has a extremely confusing package management and encapsulation.
But I wanted to adapt and improve the overall security of my system by using silverblue.
I found that other tools essentially uses podman so might as well directly uses podman.
However,from my few attempt,I found that most tutorial(official or non official) only shows how to download online containers and run it.(be it podman or toolbox)
But what I need essentially is just make sure my setup works with these conditions
- locally(minimal network/port usage if have It should have a clear documentation)
- portably(move around disk and minimal command change maybe 2 character change except the path replacement)
- transparently(like Atomic CRUD)
- fault tolerant(external reason,eg power loss)
- cli,scriptable,lts level backward compatibility,so wont need to edit scripts every (day/week/month)(some would say its a weird condition,but look at bash,copy a proper script 10 years ago,99.99% works).
- reproducible setup

So,by following existing online tutorial I found that.
______
- locally is possible(grey box workings from my exp,i dont even know if its possible to create a image locally with minimal network,its either kinda hard to find infos or I need to read source code)
- portably(seems yes,but I dont know where is it stored,and if I move it away what is the flag to use it etc,how many script i need to change)
- transparently(seems yes,foss,I can change the log level,i however cannot really know all of the container details)
- fault tolerant(I can only hope it is,else I would have corrupted images if suddenly loss power,note my older dev env is quite fault tolerant,so its a requirement)
- cli(yes,so it scriptable great,OCI complaint and can only hope its consistent)
- reproducible setup(maybe yes,seems like some setup is needed if rootless,https://github.com/containers/podman/blob/main/docs/tutorials/rootless_tutorial.md)

So with these analysis I thought why not just give it a try and ignore the fears by just doing a backup.So...

There are 2 ways I can think of to migrate.which is better or possible?
1)The braindead way*:Just create a new local podman container targeting some os,and just search how to move folder into podman in SE/SO and copy paste modify command ,run,and then try to shutdown the container and move it into my disk/thumbdrive,instead of some obsure place.

2)The smartway*:Create a minimal container then create a CI/scripts to download most dependency and then copy some stuff and archive my old dev workflow.

**Comment by baude** [comment]
I think #2 is the most compelling when using *any* container approach.  I would seriously consider putting stuff into something like a git repo including a Containerfile to perform all the setup.  I would then wire something like quay (or whoever) to rebuild my image when the git repo has a merge.  You could then either pull the image or build it locally, etc.

I don't have a super clear of your preferred workflow but I think with containers, you have enough latitude to do whatever you want.   My two cents ...

**Comment by buyvtuygJW** [comment]
i tried to use podman and i created a container.
however i dont know how to stop, it and it seems like a container created by toolbox cannot be stopped by podman?

Also how do I find out where is the container created stored?

I tried>
@fedora Xyz]$ podman stop -a
07784000404bd3348ae4171f414c69710f3f470340d0a3c4f396b8268c1ee4a6
fa66d5e7fe95dd7c650a1be397964cdcb356bbd1a16aeeabc2d67542d047594b
[a@fedora Xyz]$ podman rmi 2110dbbc33d2
Error: Image used by 07784000404bd3348ae4171f414c69710f3f470340d0a3c4f396b8268c1ee4a6: image is in use by a container

extra log:
$ podman pod stop -a
$ podman rmi 2110dbbc33d2
Error: Image used by 07784000404bd3348ae4171f414c69710f3f470340d0a3c4f396b8268c1ee4a6: image is in use by a container
$ podman ps -ap
CONTAINER ID  IMAGE                                         COMMAND               CREATED                 STATUS      PORTS       NAMES              POD ID      PODNAME
07784000404b  registry.fedoraproject.org/fedora-toolbox:36  toolbox --log-lev...  25 hours ago            Created                 fedora-toolbox-36              
fa66d5e7fe95  registry.fedoraproject.org/fedora:latest      ls                    Less than a second ago  Created                 busy_mcnulty
